chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -1909,95 +1882,73 @@ public void setTargetSpp(int value) {
    * new canvas size is not identical to the current canvas size.
    */
   public synchronized void setCanvasSize(int canvasWidth, int canvasHeight) {
-    int newWidth = Math.max(MIN_CANVAS_WIDTH, canvasWidth);
-    int newHeight = Math.max(MIN_CANVAS_HEIGHT, canvasHeight);
-    if (newWidth != width || newHeight != height) {
-      width = newWidth;
-      height = newHeight;
+    if (canvasConfig.setSize(canvasWidth, canvasHeight)) {
       initBuffers();
       refresh();
     }
   }
 
-  public synchronized void setCanvasCropSize(int canvasWidth, int canvasHeight, int fullWidth, int fullHeight, int cropX, int cropY) {
-    canvasWidth = Math.max(MIN_CANVAS_WIDTH, canvasWidth);
-    canvasHeight = Math.max(MIN_CANVAS_HEIGHT, canvasHeight);
-    if (fullWidth == 0 || fullHeight == 0) {
-      // Crop disabled
-      fullWidth = 0;
-      fullHeight = 0;
-      cropX = 0;
-      cropY = 0;
-    } else {
-      // Crop enabled
-      fullWidth = Math.max(canvasWidth(), fullWidth);
-      fullHeight = Math.max(canvasHeight(), fullHeight);
-      cropX = QuickMath.clamp(cropX, 0, fullWidth-canvasWidth());
-      cropY = QuickMath.clamp(cropY, 0, fullHeight-canvasHeight());
-    }
-    boolean changed = false;
-    if (fullWidth != this.fullWidth || fullHeight != this.fullHeight || cropX != this.cropX || cropY != this.cropY) {
-      changed = true;
-      this.fullWidth = fullWidth;
-      this.fullHeight = fullHeight;
-      this.cropX = cropX;
-      this.cropY = cropY;
-    }
-    if (canvasWidth != this.width || canvasHeight != this.height) {
-      changed = true;
-      this.width = canvasWidth;
-      this.height = canvasHeight;
+  public synchronized void setCanvasCropSize(
+    int canvasWidth, int canvasHeight,
+    int fullWidth, int fullHeight,
+    int cropX, int cropY
+  ) {
+    boolean changedBuffers = canvasConfig.setSize(canvasWidth, canvasHeight);
+    boolean cropChanged = canvasConfig.setCropSize(fullWidth, fullHeight, cropX, cropY);
+    if(changedBuffers) { 
       initBuffers();
     }
-    if (changed) {
+    if(changedBuffers || cropChanged) {
       refresh();
     }
   }
 
-  public boolean isCanvasCropped() {
-    return fullWidth != 0 && fullHeight != 0;
-  }
-
   /**
-   * @return Canvas width
+   * @deprecated use {@link CanvasConfig}
    */
+  @Deprecated
```

Comment:
They were reinserted by request from @ThatRedox [see here](https://discord.com/channels/541221265512464394/545374333883777037/1138503126509420595).
Plugins depending on the fields or getters have to be updated either way, but I would have not problem with keeping them for another version before removing them.

Reply:
I wonder if we should add a comment to deprecated methods to state a removal version? That way we would know when to remove them. Otherwise, we could attempt to track down all plugins that use said method and track compliance via an issue?

Reply:
Let's keep these methods for the 2.5.x life cycle then.